From bca80bcee7681deea6941322cd1e599d8e84f321 Mon Sep 17 00:00:00 2001
From: corebee <corebee@Corebee-Studio.localdomain>
Date: Sat, 29 Aug 2026 01:00:43 +0200
Subject: [PATCH 3/7] 5D4: track LiveView state

---
 src/propvalues.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/propvalues.c b/src/propvalues.c
index bcad34c19..c8596034a 100644
--- a/src/propvalues.c
+++ b/src/propvalues.c
@@ -234,7 +234,12 @@ PROP_HANDLER(PROP_VIDEO_MODE)
     #endif
 }
 
-#ifdef CONFIG_LIVEVIEW
+/*
+ * 5D4 does not currently enable CONFIG_LIVEVIEW globally,
+ * but it still needs PROP_LV_ACTION to keep 'lv' in sync
+ * with Canon's actual LiveView state.
+ */
+#if defined(CONFIG_LIVEVIEW) || defined(CONFIG_5D4)
 PROP_HANDLER( PROP_LV_ACTION )
 {
     lv = !buf[0];
